lib/spack/docs/packaging_guide.rst

```diff
@@ -2775,6 +2775,254 @@ Suppose a user invokes ``spack install`` like this:
 Spack will fail with a constraint violation, because the version of
 MPICH requested is too low for the ``mpi`` requirement in ``foo``.
 
+.. _custom-attributes:
+
+------------------
+Custom attributes
+------------------
+
+Often packages will need to provide attributes for dependents to query
+various details about what it provides.  While any number of custom defined
+attributes can be implemented by a pacakage, the specific four attributes
+described below are always available on every package with default
+implementations and the ability to customize with alternate implementations
+in the case of virtual packages provided:
+
+=========== =========================================== =====================
+Attribute   Purpose                                     Default
+=========== =========================================== =====================
+``root``    The installation directory for the package  ``spec.prefix``
+``command`` An executable command for the package       | ``spec.name`` in
+                                                        | ``spec.prefix.bin``
```

Comment:
Why a single command, rather than a list? Also, need to check the code blow, but is `spec.command` supposed to be a Boolean?

Reply:
Ok, I see this is documenting the already existing `command` property. The default of:
```python
spec.name in spec.prefix.bin
```
is rather confusing, since the return type of that expression is a Boolean.